### PR TITLE
Add different validation for ether and token

### PR DESCRIPTION
**Pull request by cehali**
# Summary
Add different validation for ether and token

### CODE REVIEW

File: universal-login-commons/lib/core/utils/isValidAmount.ts [hunk 1]
```diff
@@ -1,14 +1,23 @@
...
-    }
+export const isValidAmount = (tokenAddress: string, balance: Nullable<string>, gasCostInWei: utils.BigNumber, amount?: string): boolean => {
+  if (balance) {
+    const maxEtherAmount = utils.parseEther(balance).sub(gasCostInWei);
+    return tokenAddress === ETHER_NATIVE_TOKEN.address
+      ? isAmountInConstraints(maxEtherAmount, amount)
+      : isAmountInConstraints(utils.parseEther(balance), amount);
+  }
+  return false;
+};
```
Comment (marekkirejczyk) [inline universal-login-commons/lib/core/utils/isValidAmount.ts]:
```ts
export const isValidAmount = (
   tokenAddress: string, balance: Nullable<string>, 
   gasToken: string,
   gasCostInWei: utils.BigNumber, 
   amount?: string): boolean => {

if (!balance) {
  return false;
}
const maxEtherAmount = utils.parseEther(balance).sub(gasCostInWei);
if (tokenAddress === gasToken) {
      return isAmountInConstraints(maxEtherAmount, amount)
} 
return isAmountInConstraints(utils.parseEther(balance), amount);
```

File: universal-login-commons/lib/core/utils/isValidAmount.ts [hunk 2]
```diff
@@ -1,14 +1,23 @@
...
+
+export const isAmountInConstraints = (balance: utils.BigNumber, amount?: string): boolean => {
+  if (amount && amount.match(isNumber)) {
+    const amountAsBigNumber = utils.parseEther(amount);
+    return amountAsBigNumber.gt(0) && amountAsBigNumber.lt(balance);
```
Comment (marekkirejczyk) [inline universal-login-commons/lib/core/utils/isValidAmount.ts]: lt -> lte

File: universal-login-react/src/ui/Transfer/Transfer.tsx [hunk 3]
```diff
@@ -24,27 +24,20 @@ interface ErrorsProps {
   recipientError: boolean;
 }
 
-export const Transfer = ({deployedWallet, transferDetails, updateTransferDetailsWith, tokenDetails, onSendClick, transferClassName}: TransferProps) => {
+export const Transfer = ({deployedWallet, transferDetails, updateTransferDetailsWith, tokenDetailsWithBalance, tokenDetails, onSendClick, transferClassName}: TransferProps) => {
```
Comment (marekkirejczyk) [inline universal-login-react/src/ui/Transfer/Transfer.tsx]: We can now simplify and simply pass the transferService to Transfer component.
This way we can get rid of `deployedWallet`, `tokenDetails`, `tokenDetailsWithBalance `, `onSendClick` (inline), `transferDetails` (move from props to state), updateTransferDetailsWith (move to current component).

File: universal-login-react/src/ui/Transfer/Transfer.tsx [hunk 4]
```diff
@@ -24,27 +24,20 @@ interface ErrorsProps {
...
-    areInputsValid(errorsCalculated) && onSendClick();
+  const onTransferClick = async () => {
+    try {
+      await onSendClick();
+    } catch (error) {
+      setErrors({
+        amountError: error.errorType !== 'InvalidAddressOrEnsName' && true,
+        recipientError: error.errorType !== 'InvalidAmount' && true,
+      });
+    }
```
Comment (marekkirejczyk) [inline universal-login-react/src/ui/Transfer/Transfer.tsx]: We can now move ErrorProps to app layer and use them as a state instead as props.
That way we can have:
```ts
const {errors} = await transferService.transfer(transferDetails);
if (errors) {
  setErrors(errors);
} else {
  modalService.showModal('transfer');
}
```

or even better:
```ts
  history.push('/transfer')
```

Comment (xitronix) [reply universal-login-react/src/ui/Transfer/Transfer.tsx]: I planned change whole transfer flow into /transfer route and flow like in BackupCodes after this pr

**Pull request merged**